test/e2e/framework/job/wait.go

```diff
@@ -118,38 +134,52 @@ func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName s
 
 // WaitForJobFailed uses c to wait for the Job jobName in namespace ns to fail
 func WaitForJobFailed(c clientset.Interface, ns, jobName string) error {
-	return wait.PollImmediate(framework.Poll, JobTimeout, func() (bool, error) {
-		curr, err := c.BatchV1().Jobs(ns).Get(context.TODO(), jobName, metav1.GetOptions{})
+	return framework.Gomega().Eventually(context.TODO(), framework.HandleRetry(func(ctx context.Context) (*batchv1.Job, error) {
+		job, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
 		if err != nil {
-			return false, err
+			return nil, fmt.Errorf("getting the job in in namespace %s: %w", ns, err)
 		}
+		return job, nil
+	})).WithTimeout(JobTimeout).WithPolling(framework.Poll).Should(framework.MakeMatcher(func(j *batchv1.Job) (func() string, error) {
 
-		return isJobFailed(curr), nil
-	})
+		if isJobCompleted(j) {
+			return func() string {
+				return fmt.Sprintf("job %q completed while waiting for its failure", jobName)
+			}, nil
+		}
+
+		if isJobFailed(j) {
+			return nil, nil
+		}
+
```

Comment:
nit (sorry for nitpicking), but I don't see a need for the empty lines here - they are sometimes useful for readability in large functions, but I think in case of the functions in this diff they don't improve that much, and it is not clear why some places have and some don't have an empty line.

Reply:
sure! no problem, all remove them 😊